packages/plugin-commands-installation/test/update/update.ts

```diff
@@ -191,3 +191,48 @@ test('update should work normal when set empty string version', async () => {
   expect(lockfile.devDependencies['@pnpm.e2e/foo']).toEqual('2.0.0')
   expect(lockfile.devDependencies['@pnpm.e2e/peer-c']).toEqual('2.0.0')
 })
+
+test('ignore packages in package.json > pnpm.update.ignore fields in update command', async () => {
+  await addDistTag({ package: '@pnpm.e2e/foo', version: '2.0.0', distTag: 'latest' })
+  await addDistTag({ package: '@pnpm.e2e/bar', version: '100.0.0', distTag: 'latest' })
+
+  const project = prepare({
+    dependencies: {
+      '@pnpm.e2e/foo': '2.0.0',
+      '@pnpm.e2e/bar': '100.0.0',
+    },
+    pnpm: {
+      update: {
+        ignore: [
+          '@pnpm.e2e/foo',
+        ],
+      },
+    },
+  })
+
+  await install.handler({
+    ...DEFAULT_OPTS,
+    dir: process.cwd(),
+    workspaceDir: process.cwd(),
+  })
+
+  const lockfile = await project.readLockfile()
+
+  expect(lockfile.packages['/@pnpm.e2e/foo/2.0.0']).toBeTruthy()
+  expect(lockfile.packages['/@pnpm.e2e/bar/100.0.0']).toBeTruthy()
+
+  await addDistTag({ package: '@pnpm.e2e/foo', version: '100.0.0', distTag: 'latest' })
+  await addDistTag({ package: '@pnpm.e2e/bar', version: '100.1.0', distTag: 'latest' })
+
+  await update.handler({
+    ...DEFAULT_OPTS,
+    dir: process.cwd(),
+    latest: true,
+    workspaceDir: process.cwd(),
```

Comment:
This is not related to this PR, but all tests in `update/update`, which should testify `update`, not `recursive`, testify `recursive` because `workspaceDir` is defined. Is this expected behavior?

Reply:
probably not expected.

Reply:
issued: #5410 I will attack this issue later.